.github/workflows/prbot.yml

```diff
@@ -115,14 +115,30 @@ jobs:
       - uses: actions/checkout@v3
       - uses: ./.github/actions/setup
       - uses: ./.github/actions/e2e-stencil
-
-  is-valid:
-    name: 'Confirm build is valid'
-    runs-on: ubuntu-latest
+  required-jobs:
+    if: success()
     needs:
+      - 'build'
       - 'lint-check'
       - 'unit-test'
       - 'e2e-atomic-test'
       - 'e2e-quantic-test'
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo 'All required jobs have passed'
+  is-valid:
+    if: always()
+    name: 'Confirm build is valid'
+    needs:
+      - 'required-jobs'
+    runs-on: ubuntu-latest
     steps:
-      - run: echo 'Build has been verified'
+      - run: |
+          success="${{ needs.required-jobs.result == 'success' }}"
```

Comment:
the required jobs will be either "success" (all required jobs have passed) or "skipped" (required jobs did not all passed)

we basically have to do this in 2 steps